reckless_github/src/repository.rs

```diff
@@ -43,82 +50,47 @@ impl Github {
     /// related to the plugins
     pub async fn index_repository(&mut self) -> Result<(), RecklessError> {
         let repo_path = &self.url.path_string;
-        let pattern = format!("{}/[!.]*/*", &repo_path);
-
-        // FIXME rewrite it in a way that is more clear that
-        // we are walking all the plugins.
-        // for plugin_dir in repo_pat:
-        //    let conf = None
-        //    let plugin = None
-        //    for file in plugin_dir:
-        //
-        //     let lang = match file {
-        //          ...
-        //      }
-        //      plugun = {conf, lang ...}
-        //      self.plugins(plugin)
-        for plugin in glob(&pattern).unwrap() {
-            match plugin {
-                Ok(path) => {
-                    let (path_to_plugin, plugin_name) =
-                        get_plugin_info_from_path(path.clone()).unwrap();
+        for plugin_dir in WalkDir::new(repo_path)
+            .max_depth(1)
+            .into_iter()
+            .filter_entry(|dir_entry| !is_hidden(dir_entry))
```

Comment:
```suggestion
        let target_dirs = WalkDir::new(repo_path)
            .max_depth(1)
            .into_iter()
            .filter_entry(|dir_entry| !is_hidden(dir_entry));
        for plugin_dir in target_dirs
```

Reply:
we do not need to abuse of a language feature, so I think in this way it is much cleaner